src/Parser.php

```diff
@@ -640,17 +616,8 @@ protected function parseLanguage()
     {
         $token = $this->expectTokenType('Language');
 
-        if ($this->languageSpecifierLine === null) {
-            $this->lexer->analyse($this->input, $token['value']);
-            $this->languageSpecifierLine = $token['line'];
-        } elseif ($token['line'] !== $this->languageSpecifierLine) {
-            throw new ParserException(sprintf(
-                'Ambiguous language specifiers on lines: %d and %d%s',
-                $this->languageSpecifierLine,
-                $token['line'],
-                $this->file ? ' in file: ' . $this->file : ''
-            ));
-        }
+        // Switch the lexer to the specified language and tell it to treat any further language tags as comments
+        $this->lexer->analyse($this->input, $token['value'], true);
```

Comment:
Theoretically, if someone had injected a Lexer of their own which does not recognise the new `$hasSeenLanguageTag = false` then this could cause an infinite loop.

I'm not sure how much of a risk that is, and how best to avoid it.

Our own Lexer *cannot* now return a second language tag when used with our own Parser (anything after the first one will just be a comment) - so we could keep some state tracking in the Parser so that it still throws on a second language tag?

Reply:
Yes, keeping a tracker so that we can throw on a second language tag seems good. Otherwise people might expect that having two language tags should work and might be left wondering when nothing happened